core/lib/db_connection/src/lib.rs

```diff
@@ -90,7 +92,7 @@ impl TracedConnections {
     }
 }
 
-struct PooledStorageProcessor<'a> {
+pub struct PooledStorageProcessor<'a> {
```

Comment:
It's worrisome that these internal structs need to be made public; looks like inappropriate abstraction boundaries. As a benchmark, this crate should export the least amount of types possible; ideally, only `ConnectionPool` / `StorageProcessor` equivalents, and helpers like macros, instrumentation etc.